### PR TITLE
Migrations: Honor [Column(Order)] in CreateTable operation

### LINKED ISSUE #10059: Allow defining column order via [ColumnAttribute.Order] when creating tables
#2272 matched the CLR reflection order for columns within a table.

During design, we decided to wait for more feedback before honoring the value of `ColumnAttibute.Order` on the properties.

If the implementation of #2272 is insufficient for you and specifying something like `[Column(Order = 1)]` would help, please vote for this issue and add details about your scenario (if not already listed) below.

### PR DESCRIPTION
**Pull request by bricelam**
Resolves #10059

### CODE REVIEW

**Comment by bricelam** [comment]
Since SQLite does a table rebuild for *any* AlterColumn operation, we end up doing a full rebuild if you change the explicit order of a column. 😁

File: src/EFCore.Relational/Properties/RelationalStrings.resx [hunk 1]
```diff
@@ -500,6 +504,10 @@
     <value>Disposing transaction.</value>
     <comment>Debug RelationalEventId.TransactionDisposed</comment>
   </data>
+  <data name="LogDuplicateColumnOrders" xml:space="preserve">
+    <value>The configured column orders for the table '{table}' contains duplicates. Ensure the specified column order values are distinct.</value>
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Properties/RelationalStrings.resx:508]: Could include the column names

File: src/EFCore.Relational/Infrastructure/RelationalModelValidator.cs [hunk 2]
```diff
@@ -738,6 +738,23 @@ private static bool IsIdentifyingPrincipal(IEntityType dependentEntityType, IEnt
...
+                    continue;
+                }
+
+                if (!columnOrders.Add(columnOrder.Value))
+                {
+                    logger.DuplicateColumnOrders(storeObject);
+
+                    break;
+                }
+            }
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Infrastructure/RelationalModelValidator.cs:757]: Also validate in the method below that conflicting ordinals are not specified

**Review by AndriySvyryd — APPROVED** [review]

File: src/EFCore.SqlServer/Migrations/SqlServerMigrationsSqlGenerator.cs [hunk 3]
```diff
@@ -244,6 +245,11 @@ protected override void Generate(AddCheckConstraintOperation operation, IModel?
             Check.NotNull(operation, nameof(operation));
             Check.NotNull(builder, nameof(builder));
 
+            if (operation[RelationalAnnotationNames.ColumnOrder] != operation.OldColumn[RelationalAnnotationNames.ColumnOrder])
+            {
+                Dependencies.MigrationsLogger.ColumnOrderIgnoredWarning(operation);
+            }
```
Comment (bricelam) [inline src/EFCore.SqlServer/Migrations/SqlServerMigrationsSqlGenerator.cs:251]: Providers, you may also want to add this code to your AlterColumn code. (and AddColumn if you don't call base)

**Pull request merged**